pages/index.js

```diff
@@ -1,17 +1,22 @@
-import dynamic from "next/dynamic";
-import {Provider as ReduxProvider} from "react-redux";
+import { Provider as ReduxProvider } from "react-redux";
 import store from "../store";
-
-const DynamicMap = dynamic(() => import("./Map"), {
-  ssr: false,
-});
+import GridLayout from "./GridLayout";
+import FileUpload from "./FileUpload";
+import React, { useState } from 'react';
 
 export default function Home() {
+  const [data, setData] = useState([]);
+
+  const handleFileUpload = (parsedData) => {
+    setData(parsedData);
+  };
```

Comment:
Same as above, can you think about a solution using Reducer/Action in redux? see: https://redux.js.org/tutorials/fundamentals/part-3-state-actions-reducers